indexer_worker/indexer_worker/tasks.py

```diff
@@ -15,6 +16,14 @@ def _time_fmt(timestamp: int) -> str | None:
         return None
     return str(datetime.datetime.utcfromtimestamp(timestamp))
 
+@dataclass
+class TaskInfo:
+    task: Any
+    start_time: Any
+    model: str
+    target_index: str
+    finish_time: Any
+    progress: Any
```

Comment:
The types can be more precise here. Below, the times are passed to the ` _time_fmt` function that accepts integers, and probably `progress` is also an integer; @stacimc can confirm.
```suggestion
    task: Any
    start_time: int
    model: str
    target_index: str
    finish_time: int
    progress: Any
```

Reply:
Hi @krysal , `finish_time` and `progress` has the same type: <class 'multiprocessing.sharedctypes.Synchronized'> . 
But the value of `progress` in the api response is `100.0`. So should it be set as a `float` ?

Reply:
`float` would be appropriate for the timestamps (and you can change the type of the `timestamp` parameter in `_time_fmt` to match, actually!)

`progress` and `finish_time` are shared variables using `multiprocessing.Value` so I believe the correct type would be `Synchronized[float]`

Reply:
HI @stacimc When using `Synchronized[float]` , I am getting a  `TypeError: type 'Synchronized' is not subscriptable` because `Synchronized` is not a type that can be parameterized in the `typing module`. So I have used `Synchronized` as type of `progress` and `finish_time`. Also used 'float` as type for timestamps including for the parameter in  `_time_fmt_ `function. 
Have made a new commit with these changes.

Reply:
Oh, interesting! It looks like `Synchronized` is in the typeshed package/not available at runtime. I think the fix should be to add `from __future__ import annotations` to the top of the file.

Reply:
Great!
Learned something new :)